internal/processor/parser/infra.go

```diff
@@ -31,109 +32,115 @@ func NewInfraParser(
-		Value:     value,
-	})
-
-	var names []string
-	// Basic metric name
-	names = append(names, name)
-	// Legacy metric name
-	names = append(names, envelope.GetOrigin()+"."+name)
-	// BOSH alias metric name
-	if strings.HasPrefix(name, "bosh-hm-forwarder") {
-		names = append(names, strings.Replace(name, "bosh-hm-forwarder", "bosh.healthmonitor", 1))
-	}
-
-	// Create metric for each names with same values
-	for i := 0; i < len(names); i++ {
-		metrics = append(metrics, metric.MetricPackage{
-			MetricKey: &metric.MetricKey{
-				EventType: eventType,
-				Name:      names[i],
-				TagsHash:  tagsHash,
-			},
-			MetricValue: &metricValues,
+	for name, value := range getValues(envelope) {
+		// create metricValues
+		metricValues := metric.MetricValue{}
+		metricValues.Host = host
+		metricValues.Tags = tags
+		metricValues.Points = append(metricValues.Points, metric.Point{
+			Timestamp: envelope.GetTimestamp() / int64(time.Second),
+			Value:     value,
 		})
+
+		var names []string
+		// Basic metric name
+		names = append(names, name)
+		// Legacy metric name
+		if origin, ok := envelope.GetTags()["origin"]; ok {
+			names = append(names, origin+"."+name)
+		}
+		// BOSH alias metric name
+		if strings.HasPrefix(name, "bosh-hm-forwarder") {
+			names = append(names, strings.Replace(name, "bosh-hm-forwarder", "bosh.healthmonitor", 1))
+		}
+
+		// Create metric for each names with same values
+		for i := 0; i < len(names); i++ {
+			metrics = append(metrics, metric.MetricPackage{
+				MetricKey: &metric.MetricKey{
+					Name:      names[i],
+					TagsHash:  tagsHash,
+				},
+				MetricValue: &metricValues,
+			})
+		}
 	}
 
 	return metrics, nil
 }
 
-func getName(envelope *events.Envelope) string {
-	switch envelope.GetEventType() {
-	case events.Envelope_ValueMetric:
-		return envelope.GetValueMetric().GetName()
-	case events.Envelope_CounterEvent:
-		return envelope.GetCounterEvent().GetName()
-	default:
-		panic("Unknown event type")
-	}
-}
-
-func getValue(envelope *events.Envelope) float64 {
-	switch envelope.GetEventType() {
-	case events.Envelope_ValueMetric:
-		return envelope.GetValueMetric().GetValue()
-	case events.Envelope_CounterEvent:
-		return float64(envelope.GetCounterEvent().GetTotal())
+func getValues(envelope *loggregator_v2.Envelope) map[string]float64 {
+	values := map[string]float64{}
+	switch envelope.GetMessage().(type) {
+	case *loggregator_v2.Envelope_Gauge:
+		for k, v := range envelope.GetGauge().GetMetrics() {
+			values[k] = v.Value
+		}
+	case *loggregator_v2.Envelope_Counter:
+		values[envelope.GetCounter().GetName()] = float64(envelope.GetCounter().GetTotal())
 	default:
 		panic("Unknown event type")
 	}
+	return values
 }
 
 func parseTags(
-	envelope *events.Envelope,
+	envelope *loggregator_v2.Envelope,
 	environment string,
 	deploymentUUIDRegex *regexp.Regexp,
 	jobPartitionUUIDRegex *regexp.Regexp) []string {
-	tags := appendTagIfNotEmpty(nil, "deployment", envelope.GetDeployment())
```

Comment:
Where did this logic go?

Reply:
Some of these values are handled explicitly (see the rest of this function) and the rest of them moved to tags in the envelope object, so they will get handled by the loop below that ranges over all envelope tags and adds them as DD tags. [1] is a useful resource to review this part of the code.

[1] https://github.com/cloudfoundry/loggregator-api#tags

Reply:
ok! as long as we are not missing passengers 😄